packages/react/src/auto/mui/MUIFormInput.tsx

```diff
@@ -0,0 +1,132 @@
+import type { TextFieldProps } from "@mui/material";
+import { Autocomplete, Checkbox, FormControl, FormControlLabel, FormGroup, FormHelperText, TextField } from "@mui/material";
+import { DatePicker, DateTimePicker } from "@mui/x-date-pickers";
+import type { ReactElement } from "react";
+import React from "react";
+import type { Control } from "react-hook-form";
+import { useController } from "react-hook-form";
+import type { GadgetEnumConfig, GadgetFieldType } from "../../internal/gql/graphql.js";
+import type { FieldMetadata } from "../../metadata.js";
+import { FieldType } from "../../metadata.js";
+import { MUIFileInput } from "./MUIFileInput.js";
+import { MUIJSONInput } from "./MUIJSONInput.js";
+import { MUIRolesCombobox } from "./MUIRolesCombobox.js";
+
+const FieldTypeToInputType: Partial<Record<GadgetFieldType, TextFieldProps["type"]>> = {
+  [FieldType.Number]: "number",
+  [FieldType.Email]: "email",
+  [FieldType.Password]: "password",
+  [FieldType.EncryptedString]: "password",
+};
+
+export const MUIAutoFormControl = (props: { path: string; control: Control<any>; field: FieldMetadata; children: ReactElement }) => {
```

Comment:
`field: FieldMetadata` gets confusing where we have other places where we have `field: string` for the string api id. Consider renaming this to `fieldMetadata: FieldMetadata`